docs/source/emb_sim.rst

```diff
@@ -0,0 +1,106 @@
+This task is to find the most similar entity/relation embeddings for some pre-defined similarity functions given a set of entities or relations. An example of the output for top5 similar entities are as follows::
+
+    left     right    score
+    0        0        0.99999
+    0        18470    0.91855
+    0        2105     0.89916
+    0        13605    0.83187
+    0        36762    0.76978
+
+Currently we support five different similarity functions: cosine, l2 distance, l1 distance, dot product and extended jaccard.
+
+Four arguments are required to provide basic information for finding similar embeddings:
+
+  * ``--emb_file``, The numpy file that contains the embeddings of all entities/relations in a knowledge graph.
+  * ``--format``, The format of the input objects (entities/relations).
+
+    * ``l_r``: two list of objects are provided as left objects and right objects.
+    * ``l_*``: one list of objects is provided as left objects and all objects in emb\_file are right objects. This is to find most similar objects to the ones on the left.
+    * ``*_r``: one list of objects is provided as right objects list and treat all objects in emb\_file as left objects.
+    * ``*``: all objects in the emb\_file are both left objects and right objects. The option finds the most similar objects in the graph.
+
+  * ``--data_files`` A list of data file names. It provides necessary files containing the requried data according to the format, e.g., for ``l_r``, two files are required as left_data and right_data, while for ``l_*``, one file is required as left_data, and for ``*`` this argument will be omited.
+  * ``--raw_data``, A flag indicates whether the data in data_files are raw IDs or KGE IDs. If True, the data are the Raw IDs and the command will map the raw IDs to KGE Ids automatically using the ID mapping file provided through ``--mfile``. If False, the data are KGE IDs. Default: False.
+
+Task related arguments:
+
+  * ``--exec_mode``, Indicate how to calculate scores for element pairs and calculate topK. Default: 'all'
+
+    * ``pairwise``: The same number (N) of left and right objects are provided. It calculates the similarity pair by pair: result = topK([score(l_i, r_i) for i in N]) and output the K most similar pairs.
+    * ``all``: both left objects and right objects are provided as L and R. It calculates similarity scores of all possible combinations of (l_i, r_j): result = topK([[score(l_i, rj) for l_i in L] for r_j in R]), and outputs the K most similar pairs.
+    * ``batch_left``: left objects and right objects are provided as L and R. It finds the K most similar objects from the right objects for each object in L: result = topK([score(l_i, r_j) for r_j in R]) for l_j in L. It outputs (len(L) * K) most similar pairs.
+
+  * ``--topk``, How many results are returned. Default: 10.
+  * ``--sim_func``, the function to define the distance between a pair of objects. It support five functions. Default: cosine
```

Comment:
should we call them distance function or similarity function? The goal is to find similar pairs. For L2 and L1, we want to find pairs with the smallest l1 and l2 values. For cosine and dot, we try to find the pairs with largest values. is it right?